src/helm/benchmark/presentation/create_plots.py

```diff
@@ -0,0 +1,583 @@
+                value = np.nan
+            values.append(value)
+        return values
+
+    adapters: Optional[List[str]] = None
+    columns: List[Column] = []
+    mean_win_rates: Optional[np.ndarray] = None
+    for header_cell, *column_cells in zip(raw_table["header"], *raw_table["rows"]):
+        cell_values = get_cell_values(column_cells)
+        if header_cell["value"] == "Model/adapter":
+            adapters = cell_values
+        elif header_cell["value"] == "Mean win rate":
+            mean_win_rates = np.array(cell_values)
+        else:
+            assert "metadata" in header_cell
+            name = header_cell["value"]
+            group = header_cell["metadata"]["run_group"]
+            metric = header_cell["metadata"]["metric"]
+            lower_is_better = header_cell["lower_is_better"] if "lower_is_better" in header_cell else None
+            columns.append(Column(name, group, metric, np.array(cell_values), lower_is_better))
+    assert adapters is not None
+
+    return Table(adapters, columns, mean_win_rates)
+
+
+def get_color_palette(n_colors: int) -> sns.palettes._ColorPalette:
+    if n_colors < 6:
+        return sns.color_palette("colorblind", n_colors=n_colors)
+    else:
+        return sns.color_palette(colorcet.glasbey_warm, n_colors=n_colors)
+
+
+def draw_box_plot(
+    x_to_ys: Mapping[str, Union[List[float], np.ndarray]], ax: matplotlib.axis.Axis, rotate_xticklabels: bool = True
+):
+    """Given a mapping from string to floats, draw a box plit on the given axis ax."""
+    xs = []
+    all_ys = []
+    for x, ys in x_to_ys.items():
+        ys = [y for y in ys if not np.isnan(y)]
+        if ys:
+            xs.append(x)
+            all_ys.append(ys)
+            ax.scatter([len(xs)] * len(ys), ys, c="#bbb")
+    ax.boxplot(all_ys, boxprops={"linewidth": 2}, medianprops={"linewidth": 2})
+    if rotate_xticklabels:
+        ax.set_xticklabels(xs, rotation=-30, ha="left")
+    else:
+        ax.set_xticklabels(xs)
+
+
+class Plotter:
+    """
+    Main class producing plots. Each create_*() method reads group data from base_path and creates a plot at the
+    save_path. create_all_plots() runs all these functions at once.
+    """
+
+    def __init__(self, base_path: str, save_path: str):
+        self.base_path = base_path
+        self.save_path = save_path
+        self._tables_cache: Dict[str, Dict[str, Table]] = {}
+
+        schema = read_schema()
+        self.model_metadata = {model_field.display_name: model_field for model_field in schema.models}
+
+    def get_group_tables(self, group_name: str) -> Dict[str, Table]:
+        """Reads and parses group tables. Uses _tables_cache to avoid reprocessing the same table multiple times."""
+        if group_name in self._tables_cache:
+            return self._tables_cache[group_name]
+        with open(os.path.join(self.base_path, "groups", f"{group_name}.json")) as fp:
```

Comment:
Read it as the `Table` class in `table.py`?

Reply:
The structure I'm using here is a bit different and makes plotting easier/smoother. So I'd rather stick with it, unless there is a reason we want it otherwise. (we can trivially convert `table.Table` to this format with `parse_table` if we need to)